From 18a65f021735b800afc0cb811049b58cbda8858d Mon Sep 17 00:00:00 2001
From: Claudiu Manoil <claudiu.manoil@nxp.com>
Date: Wed, 30 Jan 2019 16:59:35 +0200
Subject: [PATCH 646/706] arm64: dts: fsl: ls1028a-rdb: Add ENETC external eth
 ports for the LS1028A RDB board

The LS1028A RDB board features an Atheros PHY connected over
SGMII to the ENETC PF0 (or Port0).  ENETC Port1 (PF1) has no
external connection on this board, so it can be disabled for now.

Signed-off-by: Alex Marginean <alexandru.marginean@nxp.com>
Signed-off-by: Claudiu Manoil <claudiu.manoil@nxp.com>
(cherry picked from commit b0ed3f69c34009614c500aa454c7bc72b62f51f5)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 .../boot/dts/freescale/fsl-ls1028a-rdb.dts      | 17 ++++++++++-------
 1 file changed, 10 insertions(+), 7 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1028a-rdb.dts b/arch/arm64/boot/dts/freescale/fsl-ls1028a-rdb.dts
index dc5460a0b15d..083ed62d6551 100755
--- a/arch/arm64/boot/dts/freescale/fsl-ls1028a-rdb.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1028a-rdb.dts
@@ -55,7 +55,6 @@
 		crypto = &crypto;
 		serial0 = &duart0;
 		serial1 = &duart1;
-		enetc_port0_sgmii_phy0 = &sgmii_phy0;
 	};
 
 	chosen {
@@ -244,12 +243,6 @@
 };
 
 &mdio0 {
-	sgmii_phy0: ethernet-phy@2 {
-		reg = <0x2>;
-		// Atheros PHY on RDB seems to cause traffic loss @ 1Gbps w EEE
-		eee-broken-1000t;
-	};
-
 	qsgmii_phy1: ethernet-phy@4 {
 		reg = <0x10>;
 	};
@@ -270,6 +263,16 @@
 &enetc_port0 {
 	phy-handle = <&sgmii_phy0>;
 	phy-connection-type = "sgmii";
+
+	mdio {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		sgmii_phy0: ethernet-phy@2 {
+			reg = <0x2>;
+			// Atheros PHY on RDB seems to cause traffic loss @ 1Gbps w EEE
+			eee-broken-1000t;
+		};
+	};
 };
 
 &enetc_port1 {
-- 
2.17.1

